### PR TITLE
emsdk: fixes for conan 2.x client

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

**Comment by prince-chrismc** [comment]
It does not pass the v2 pipeline fyi

```   
 -------- Computing dependency graph --------
 nodejs/16.3.0: Not found in local cache, looking in remotes...
 nodejs/16.3.0: Checking remote: conan-center-v2
 nodejs/16.3.0: Checking remote: c3i_PR-v2-13600
 Graph root
     virtual
 Requirements
     emsdk/3.1.18#82047224731c449a472b67b51d03fb58 - Cache
 Graph error
     Package 'nodejs/16.3.0' not resolved: Unable to find 'nodejs/16.3.0' in remotes
 
 -------- Computing necessary packages --------
 ERROR: Package 'nodejs/16.3.0' not resolved: Unable to find 'nodejs/16.3.0' in remotes
 ```

File: recipes/emsdk/all/conanfile.py [hunk 1]
```diff
@@ -176,16 +176,16 @@ def package_info(self):
...
-        self.env_info.RANLIB = self._define_tool_var("emranlib")
-        self.env_info.STRIP = self._define_tool_var("emstrip")
+        if Version(conan_version).major < 2:
+            self.env_info.PATH.extend(self._paths)
+            self.env_info.CONAN_CMAKE_TOOLCHAIN_FILE = toolchain
+            self.env_info.EMSDK = self._emsdk
+            self.env_info.EMSCRIPTEN = self._emscripten
+            self.env_info.EM_CONFIG = self._em_config
+            self.env_info.EM_CACHE = self._em_cache
+            self.env_info.CC = self._define_tool_var("emcc")
```
Comment (prince-chrismc) [inline recipes/emsdk/all/conanfile.py:186]: would this not be useful to have with build env?

Comment (prince-chrismc) [reply recipes/emsdk/all/conanfile.py:186]: found it https://github.com/conan-io/conan-center-index/pull/13600/files#diff-0f83176c5b3c7cd12cc71869c6204c8c2d6600e3f97d518f984edbe12d63e928R163

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Since this is not used by anything... couldnt we just delete the old ones?

https://conan.io/center/emsdk?tab=usedby

**Pull request merged**